src/providers/ogr/qgsogrdataitems.cpp

```diff
@@ -388,14 +390,28 @@ QgsOgrDataCollectionItem::QgsOgrDataCollectionItem( QgsDataItem *parent, const Q
 QVector<QgsDataItem *> QgsOgrDataCollectionItem::createChildren()
 {
   QVector<QgsDataItem *> children;
+  QStringList skippedLayerNames;
 
   gdal::dataset_unique_ptr hDataSource( GDALOpenEx( mPath.toUtf8().constData(), GDAL_OF_VECTOR, nullptr, nullptr, nullptr ) );
+  GDALDriverH hDriver = GDALGetDatasetDriver( hDataSource.get() );
+  QString driverName = QString::fromUtf8( GDALGetDriverShortName( hDriver ) );
+  if ( driverName == QStringLiteral( "SQLite" ) )
+  {
+    char **papszOptions = nullptr;
+    papszOptions = CSLSetNameValue( papszOptions, "LIST_ALL_TABLES", "YES" );
```

Comment:
could this be safely set for all drivers, to avoid the double-open?

Reply:
If our minimum GDAL version requirement is 2.1, we could use "@LIST_ALL_TABLES" (the @ prefix was added in that version). Prior to that, passing on an option to a driver that doesn't support it outputs a warning. Not sure whether the warning is harmful in any way since I have GDAL 2.3 on my machine. That's my understanding of things ATM :)

I've tested this with a large 2gb sqlite database, the double-open didn't create any delays that I could see.